### DISCUSSION TITLE
Named arguments in function snippet

### DISCUSSION THREAD

**Discussion by sellithy** [microsoft/pyright#9203] [Q&A] [answered]
If I have
```
def foo(param1: float, param2: float): ...
```
I want to type `foo<TAB>` and get
```
foo(
    param1=$1,
    param2=$2,
)$0
```
1. Does Pyright support this right now? (I think I've done enough research to say no but want to double check)
2. If not, would this be a good addition to Pyright?
3. If yes, how hard would it be to implement and can I get some pointers on how to implement it?

**Comment by erictraut** [comment, marked as answer]
No, this isn't supported in pyright currently. I can say that personally I would find this pretty annoying and unintuitive, especially for parameters that support both positional and keyword arguments. Do you normally provide keyword arguments for all parameters? I think that's pretty far from the norm in Python code.

In any case, pyright is focused on type checking, and what you're proposing here is a language server feature. While pyright has some basic language server functionality, we have no plans to add new language server features to pyright. If you'd like to see new language server functionality in pylance (the VS Code language server that's built on top of pyright), you could submit a feature request to Microsoft's pylance team [here](https://github.com/microsoft/pylance-release/issues). For ideas like this, they typically wait for signal (thumbs-up reactions) from other pylance users to see if it's a broadly-desired feature.

**Comment by sellithy** [comment]
Thank you for the quick response :)

> I think that's pretty far from the norm in Python code

I should've clarified, I wanted this as a configurable option or as a secondary generated snippet, definitely not the default. I am not a psycho that always names my parameters lol

> what you're proposing here is a language server feature

This is my fault but I thought Pyright was a full LSP and not just the type checker (although it clearly says that in the README 😭)
This is purely FYI but part of my confusion is because I use [coc-pyright](https://github.com/fannheyward/coc-pyright) and I just _assumed_ that Pyright would be an LSP (again, fully my fault)
